pkg/registry/apis/provisioning/repository/github.go

```diff
@@ -896,6 +901,10 @@ func (r *githubRepository) OnDelete(ctx context.Context) error {
 	return r.deleteWebhook(ctx)
 }
 
+func (r *githubRepository) Clone(ctx context.Context, opts CloneOptions) (ClonedRepository, error) {
```

Comment:
This is a way for us to plug the gogit behaviour into the GithubRepository without creating interdependencies. 

Github repository can now offer the cloning functionality without extra code.